app/src/main/java/com/example/githubclone/presentation/qrscanner/QRCodeAnalyzer.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.githubclone.presentation.qrscanner
+
+import android.annotation.SuppressLint
+import androidx.camera.core.ImageProxy
+import com.google.mlkit.vision.barcode.BarcodeScanner
+import com.google.mlkit.vision.barcode.common.Barcode
+import com.google.mlkit.vision.common.InputImage
+import timber.log.Timber
+
+@SuppressLint("UnsafeOptInUsageError")
+fun processImageProxy(
+    barcodeScanner: BarcodeScanner,
+    imageProxy: ImageProxy,
+    onSuccess: (barcodes: List<Barcode>) -> Unit,
+) {
+    val inputImage =
+        InputImage.fromMediaImage(imageProxy.image!!, imageProxy.imageInfo.rotationDegrees)
+
+    barcodeScanner.process(inputImage)
+        .addOnSuccessListener { barcodes ->
+            onSuccess(barcodes)
+        }
+        .addOnFailureListener {
+            Timber.e(it.message.toString())
+        }.addOnCompleteListener {
+            // When the image is from CameraX analysis use case, must call image.close() on received
+            // images when finished using them. Otherwise, new images may not be received or the camera
+            // may stall.
+            imageProxy.close()
```

Comment:
Let's check if we could use `.await()` instead

Use this dependency if you can't find the await method: 
`implementation("org.jetbrains.kotlinx:kotlinx-coroutines-play-services:1.6.4")`